server/src/main/java/org/apache/iotdb/db/metadata/idtable/IDTableHashmapImpl.java

```diff
@@ -129,6 +134,42 @@ public synchronized void createTimeseries(CreateTimeSeriesPlan plan) throws Meta
     deviceEntry.putSchemaEntry(plan.getPath().getMeasurement(), schemaEntry);
   }
 
+  /**
+   * Delete all timeseries matching the given paths
+   *
+   * @param fullPaths paths to be deleted
+   * @return deletion failed Timeseries
+   * @throws MetadataException
+   */
+  @Override
+  public Pair<Integer, Set<String>> deleteTimeseries(List<PartialPath> fullPaths)
+      throws MetadataException {
+    int deletedNum = 0;
+    Set<String> failedNames = new HashSet<>();
+    List<Pair<PartialPath, Long>> deletedPairs = new ArrayList<>(fullPaths.size());
+    for (PartialPath fullPath : fullPaths) {
+      try {
+        Map<String, SchemaEntry> map = getDeviceEntry(fullPath.getDevice()).getMeasurementMap();
+        Long offset = map.get(fullPath.getMeasurement()).getDiskPointer();
+        deletedPairs.add(new Pair<>(fullPath, offset));
+      } catch (NullPointerException e) {
+        failedNames.add(fullPath.getFullPath());
+      }
```

Comment:
Replace this NullPointerException catch with ```if(xx==null)```.

Reply:
fixed